google/cloud/spanner/internal/connection_impl.cc

```diff
@@ -220,6 +247,12 @@ class StatusOnlyResultSetSource : public internal::ResultSourceInterface {
 
 class DmlResultSetSource : public internal::ResultSourceInterface {
  public:
+  static StatusOr<std::unique_ptr<ResultSourceInterface>> Create(
```

Comment:
I think `DmlResult` and `QueryResult` are quite different types and would likely best be treated separately. In a future PR we may want to remove all connections between ResultSourceInterface and DmlResult, and I think it'll be a simplification.

No change needed in this PR, but we should probably discuss if this is a good idea or not.

Reply:
ResultSetSource is an interface to reading either of the spanner protos: ResultSet or PartialResultSet. It's not specific to queries or dml operations.

Reply:
`ResultSet` is so trivial that using `ResultSetSource` interface to read it is adding complexity, IMO. But we can fix this in another PR.